apps/heft/src/plugins/ProjectValidatorPlugin.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fs from 'fs';
+import { FileSystem, LegacyAdapters } from '@rushstack/node-core-library';
+
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { Constants } from '../utilities/Constants';
+import { ScopedLogger } from '../pluginFramework/logging/ScopedLogger';
+import { IHeftPlugin } from '../pluginFramework/IHeftPlugin';
+import { HeftSession } from '../pluginFramework/HeftSession';
+import { IHeftLifecycle } from '../pluginFramework/HeftLifecycle';
+
+const ALLOWED_HEFT_DATA_FOLDER_FILES: Set<string> = new Set<string>();
+const ALLOWED_HEFT_DATA_FOLDER_SUBFOLDERS: Set<string> = new Set<string>([Constants.buildCacheFolderName]);
+
+const PLUGIN_NAME: string = 'ProjectValidatorPlugin';
+
+export class ProjectValidatorPlugin implements IHeftPlugin {
```

Comment:
It's not at all obvious what `ProjectValidatorPlugin` is supposed to do. Could you add some code comments explaining its purpose?